File: 300-longest-increasing-subsequence/300-longest-increasing-subsequence.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& a) {
        
        int n=a.size();
        
        vector<int> dp(n,1);
        
        for(int i=0;i<n;i++)
        {
            for(int j=0;j<i;j++)
            {
                if(a[i]>a[j] && dp[i]<dp[j]+1)
                {
                    dp[i]=dp[j]+1;
                }
                
            }
        }
        
        int mx=0;
        
        for(int i=0;i<n;i++)
        {
            mx=max(mx,dp[i]);
        }
        
        return mx;
    }
};

File: 300-longest-increasing-subsequence/300-longest-increasing-subsequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var lengthOfLIS = function(a) {
    
    let n=a.length;
    
    let dp=new Array();
    
    
    for(let i=0;i<n;i++)
    {
        dp[i]=1;
        
        for(let j=0;j<i;j++)
        {
            if(a[i]>a[j] && dp[i]<dp[j]+1)
            {
                dp[i]=dp[j]+1;
            }
        }
    }
    
    let mx=0;
    
    for(let i=0;i<n;i++)
    {
        mx=Math.max(mx,dp[i]);
    }
    
    return mx;
};